sdks/js/packages/core/react/components/organization/api-keys/delete.tsx

```diff
@@ -0,0 +1,104 @@
+import { Button, Flex, Text, toast } from '@raystack/apsara/v1';
+import cross from '~/react/assets/cross.svg';
+import { useNavigate, useParams } from '@tanstack/react-router';
+import { useFrontier } from '~/react/contexts/FrontierContext';
+import { useState } from 'react';
+
+export const DeleteServiceAccount = () => {
+  const { id } = useParams({ from: '/api-keys/$id/delete' });
+  const navigate = useNavigate({ from: '/api-keys/$id/delete' });
+  const { client, activeOrganization: organization } = useFrontier();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const orgId = organization?.id;
+
+  async function onDeleteClick() {
+    try {
+      setIsLoading(true);
+      await client?.frontierServiceDeleteServiceUser(id, { org_id: orgId });
+      navigate({
+        to: '/api-keys',
+        state: {
+          refetch: true
+        }
+      });
+      toast.success('Service account deleted');
+    } catch (err: any) {
+      toast.error('Unable to delete service account', {
+        description: err?.message
+      });
+    } finally {
+      setIsLoading(true);
+    }
+  }
+
+  function onCancel() {
+    navigate({ to: '/api-keys' });
+  }
+
+  return (
+    <Dialog open={true}>
+      {/* @ts-ignore */}
+      <Dialog.Content
+        overlayClassname={styles.overlay}
+        className={styles.addDialogContent}
+      >
+        <Flex justify="between" className={styles.addDialogForm}>
+          <Text size={6} weight={500}>
+            Delete Service Account
+          </Text>
+
+          <Image
+            alt="cross"
+            style={{ cursor: 'pointer' }}
+            // @ts-ignore
+            src={cross}
+            onClick={() => navigate({ to: '/api-keys' })}
+            data-test-id="frontier-sdk-delete-service-account-close-btn"
+          />
+        </Flex>
+        <Separator />
+
+        <Flex
+          direction="column"
+          gap="medium"
+          className={styles.addDialogFormContent}
+        >
+          <Text>
+            This is an irreversible and permanent action doing this might result
+            in deletion of the service account and the keys associated with it.
+            Do you wish to proceed?
+          </Text>
+        </Flex>
+        <Separator />
+        <Flex
+          justify="end"
+          className={styles.addDialogFormBtnWrapper}
+          gap={'medium'}
+        >
+          <Button
+            variant="secondary"
+            size="normal"
+            data-test-id="frontier-sdk-delete-service-account-cancel-btn"
+            onClick={onCancel}
+          >
+            Cancel
+          </Button>
+          <Button
+            variant="danger"
+            size="normal"
+            data-test-id="frontier-sdk-delete-service-account-confirm-btn"
+            loading={isLoading}
+            disabled={isLoading}
+            onClick={onDeleteClick}
+            loaderText="Deleting..."
+          >
+            I Understand and Delete
```

Comment:
I think we can reframe this sentence better?

Reply:
This is as per design
<img width="776" alt="Screenshot 2024-11-29 at 12 07 23 PM" src="https://github.com/user-attachments/assets/0489167c-a0c4-4c3d-92fc-ad91b98de791">